app/services/katello/pulp3/repository/yum.rb

```diff
@@ -70,28 +70,53 @@ def partial_repo_path
           "/pulp/repos/#{repo.relative_path}/".sub('//', '/')
         end
 
-        def copy_units(source_repository_version_href, content_unit_hrefs, dependency_solving)
+        def copy_units(source_repository, content_unit_hrefs, dependency_solving, dest_base_version = 0,
+                       additional_repo_map = {})
           tasks = []
 
           if content_unit_hrefs.any?
             data = PulpRpmClient::Copy.new
-            data.config = [
-              source_repo_version: source_repository_version_href,
+            data.config = [{
+              source_repo_version: source_repository.version_href,
               dest_repo: repository_reference.repository_href,
-              dest_base_version: 0,
+              dest_base_version: dest_base_version,
               content: content_unit_hrefs.sort
-            ]
+            }]
             data.dependency_solving = dependency_solving
+            if dependency_solving
+              # repo_map example: {
+              #   <source_repo_id>: {
+              #     dest_repo: <dest_repo_id>,
+              #     base_version: <base_version>
+              #   }
+              # }
+              additional_repo_map.each do |source_repo, dest_repo_map|
+                source_repo_version = ::Katello::Repository.find(source_repo).version_href
+
+                dest_repo = ::Katello::Repository.find(dest_repo_map[:dest_repo])
+                dest_repo_href = ::Katello::Pulp3::Repository::Yum.new(dest_repo, SmartProxy.pulp_master).repository_reference.repository_href
+                data.config << {
+                  source_repo_version: source_repo_version,
+                  dest_repo: dest_repo_href,
+                  dest_base_version: dest_repo_map[:base_version],
+                  content: content_unit_hrefs.sort
+                }
```

Comment:
Does this _need_ to be sorted?

Reply:
It was sorted before so that the VCR tests would work.  Maybe there's a way to keep that sorting on the testing side of things?

Reply:
or maybe sort outside the loop once.